### ISSUE TITLE
Incorrect rendering of tinted character in BitmapText with Canvas renderer

### ISSUE THREAD

**Issue by SBCGames** [photonstorm/phaser-ce#58]
This Issue is about

* A bug in the API (checked from 2.6.1 on)

 When using multiple tinted BitmapTexts some characters may be rendered incorrectly. It can even lead to IndexSizeError in some browsers.

 On image below is test scenario:

![bitmaptextbug](https://cloud.githubusercontent.com/assets/8962889/23823697/70f29de8-0668-11e7-9edc-f0803bd2ab56.png)

 Two BitmapTexts are used to show number 5. Then, when both texts are changed to 10, only first (bottom one) text is rendered correctly, while the second has distorted "50" instead of 10. Problem is related only to Canvas renderer.
 Problem is that both texts use the same character and parameter requiresReTint is not on glyph, but on texture. Then tinted texture for first character occurence is updated, but not for other characters sharing the same texture (as requiresReTint is flagged to false after first retint). So, second text uses old tinted texture, but tries to display correct rectangle from it. If rectangle to display is bigger than tinted texture, you may get IndexSizeError in some browsers (IE).

 Solution working for me was to change (https://github.com/photonstorm/phaser-ce/blob/master/src/gameobjects/BitmapText.js#L455):
```javascript
            g.scale.set(scale);
            g.tint = this.tint;
            g.texture.requiresReTint = true;
```
to:
```javascript
            g.scale.set(scale);
            g.tint = this.tint;
            g.texture.requiresReTint = true;
            g.cachedTint = 0xFFFFFF;
```

**Comment by DDowell** [comment]
Great explanation as to what is going on. Thank you and lets hope for a quick patch on this. For now, I use separate textures for all fonts that need tinting.

**Comment by samme** [comment]
<http://codepen.io/samme/pen/Zewjjo>

**Comment by samme** [comment]
<http://codepen.io/samme/pen/bqJgMV>

**Comment by samme** [comment]
Fixed in #105
